index.bs

```diff
@@ -2219,6 +2256,89 @@ To <dfn>parse trigger specs</dfn> given a [=map=] |map|, a [=moment=]
         1. Set |i| to |i| + 1.
 1. Return |specs|.
 
+To <dfn>parse max number of event states</dfn> from a [=map=] map,
+a source type |sourceType|,
+a possibly null 32-bit unsigned integer |attributionScopeLimit|,
+and a [=randomized response output configuration=] |config|:
+1. If |attributionScopeLimit| is null:
+    1. If |map|["`max_num_event_states`"] [=map/exists|exist=], return an error.
+    1. Return null.
+1. Let |states| be the [=obtain a set of possible trigger states|number of possible trigger states=] with |config|.
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| > 3 
+    and |map|["`max_num_event_states`"] does not [=map/exist=], return an error.
+1. Let |maxNumEventStates| be the result of running
+    [=parse an optional 32-bit unsigned integer=] with |value|,
+    "`max_num_event_states`", and 3.
+1. If |maxNumEventStates| is an error or |maxNumEventStates| is 0, return an error.
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| > |maxNumEventStates|, return an error.
+1. Return |maxNumEventStates|
```

Comment:
We should probably also prohibit the value from being greater than https://pr-preview.s3.amazonaws.com/feifeiji89/attribution-reporting-api/pull/1215.html#max-trigger-state-cardinality